main.tf

```diff
@@ -248,6 +248,8 @@ resource "aws_s3_bucket_lifecycle_configuration" "default" {
 
   bucket = aws_s3_bucket.default.bucket
 
+  transition_default_minimum_object_size = var.transition_default_minimum_object_size
```

Comment:
please remove the enter between bucket and transition_default_minimum_object_size, and run terraform fmt --recursive